### ISSUE TITLE
ServiceWorker.postMessage() references the not-yet-created worker global

### ISSUE THREAD

**Issue by jungkees** [w3c/ServiceWorker#1126]
Raised by https://github.com/w3c/ServiceWorker/pull/1102#issuecomment-291405609.

> We call "Run Service Worker", which in parallel creates the workers global object, yet postMessage in the next step tries to synchronously access that global (which hasn't been created yet at that point). It seems instead postMessage should synchronously serialize the message, and then queue a task on the worker to actually deliver the message event after the worker has started up?

**Comment by jungkees** [comment]
I think we need the global object to get the target realm to invoke `StructuredCloneWithTransfer(message, transfer, targetRealm)`. This serialization should be done on the calling context, right? Or does it make sense to run `StructuredCloneWithTransfer()` in the queued task?

**Comment by domenic** [comment]
I think the idea is that you stop using StructuredCloneWithTransfer() entirely, and instead use separate StructuredSerializeWithTransfer/StructuredDeserializeWithTransfer steps. Maybe I can work that out in a new PR.

**Comment by jungkees** [comment]
> use separate StructuredSerializeWithTransfer/StructuredDeserializeWithTransfer steps.

I see. That makes sense.

> Maybe I can work that out in a new PR.

I'd appreciate it.
